coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1441,6 +1441,36 @@ func (s *server) CompleteJob(ctx context.Context, completed *proto.CompletedJob)
 			return nil, xerrors.Errorf("complete job: %w", err)
 		}
 
+		// Insert timings outside transaction since it is metadata.
+		// nolint:exhaustruct // The other fields are set further down.
+		params := database.InsertProvisionerJobTimingsParams{
+			JobID: jobID,
+		}
+		for _, t := range completed.GetWorkspaceBuild().GetTimings() {
+			if t.Start == nil || t.End == nil {
+				s.Logger.Warn(ctx, "timings entry has nil start or end time", slog.F("entry", t.String()))
```

Comment:
nit, non-blocking: Instead of logging the whole entry, could we instead log individual fields? I don't see anything sus in the fields right now, but it could lead us to log something later that we'd rather not.

Reply:
Good call. However, I'm wondering how valuable this log entry will be without the `resource`, which is the most likely attribute which may leak data. I think Terraform is pretty good with keeping sensitive strings masked, so I'll defer to their good judgment for now. If we were running a SaaS, I'd be more paranoid.